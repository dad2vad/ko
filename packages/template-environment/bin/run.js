#!/usr/bin/env node

require("@kosko/template").run(require("../dist/index.node.cjs").default);
