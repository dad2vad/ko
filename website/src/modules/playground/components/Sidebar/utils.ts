export function isValidName(value: string): boolean {
  return /^[\w-_.]+$/.test(value);
}
