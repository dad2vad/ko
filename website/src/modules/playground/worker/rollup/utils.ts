export function getModuleURLForCDN(module: string) {
  return `https://jspm.dev/${module}`;
}
