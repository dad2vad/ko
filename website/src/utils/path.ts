export const sep = "/";

export function isAbsolute(path: string): boolean {
  return path.startsWith(sep);
}

export function isRelative(path: string): boolean {
  return path.startsWith(`.${sep}`) || path.startsWith(`../${sep}`);
}

export function dirname(path: string): string {
  const index = path.lastIndexOf(sep);
  if (index === -1) return ".";
  return path.substring(0, index) || sep;
}

export function extname(path: string): string {
  const index = path.lastIndexOf(".");
  if (index === -1) return "";
  return path.substring(index);
}

export function basename(path: string, ext?: string): string {
  const index = path.lastIndexOf(sep);
  const base = index === -1 ? path : path.substring(index + 1);

  if (base.endsWith(ext)) {
    return base.substring(0, base.length - ext.length);
  }

  return base;
}

export function relative(from: string, to: string): string {
  const fromParts = from.split(sep).filter(Boolean);
  const toParts = to.split(sep).filter(Boolean);

  while (fromParts[0] && toParts[0]) {
    fromParts.shift();
    toParts.shift();
  }

  while (toParts[0] === "." || toParts[0] === "..") {
    const toPart = toParts.shift();

    if (toPart === "..") {
      fromParts.pop();
    }
  }

  while (fromParts.pop()) {
    toParts.unshift("..");
  }

  return toParts.join(sep);
}
