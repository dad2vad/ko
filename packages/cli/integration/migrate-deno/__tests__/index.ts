import { runDenoCLI } from "../../utils/run";

test("should print migrated manifests", async () => {
  const result = await runDenoCLI(["migrate", "-f", "-"], {
    input: `apiVersion: v1
kind: Pod
metadata:
  name: test-pod
spec:
  containers: []`
  });

  expect(result.stdout).toMatchSnapshot();
});
