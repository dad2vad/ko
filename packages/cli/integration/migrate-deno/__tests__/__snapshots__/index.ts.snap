// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should print migrated manifests 1`] = `
"import { Pod } from "kubernetes-models/v1/Pod";

const pod = new Pod({
  "metadata": {
    "name": "test-pod"
  },
  "spec": {
    "containers": []
  }
});

export default [pod];"
`;
